### ISSUE TITLE
Generated schema fails with nested arrays under oneOf 

### ISSUE THREAD

**Issue by pauldzy** [swagger-api/swagger-ui#7042]
### Q&A (please complete the following information)
 - OS: Windows
 - Browser: Chrome
 - Version: 89.0.4389.72
 - Method of installation: N/A
 - Swagger-UI version: 3.44.1
 - Swagger/OpenAPI version: OpenAPI 3.0

### Content & configuration

Example Swagger/OpenAPI definition:
```yaml
openapi: 3.0.3
info:
  title: 'GeoJSON Geometry'
  version: 1.0.0

paths:
  '/data':
    get:
      responses:
        '200':
          description: 'GeoJSON'
          content:
            application/json:
              schema:
                type: object
                properties:
                  type: 
                    $ref: '#/components/schemas/geometry'

components:
  schemas:
    geometry:
      type: object
      properties:
        type:
          type: string
        coordinates:
          oneOf:
            - $ref: '#/components/schemas/point'
            - $ref: '#/components/schemas/line'
            - $ref: '#/components/schemas/polygon'
            - type: array
              title: GeoJSON BBOX
              items:
                $ref: '#/components/schemas/ordinate'
      required:
        - type
        - coordinates
    point:
      type: array
      items:
        $ref: '#/components/schemas/ordinate'
    line:
      type: array
      items:
        $ref: '#/components/schemas/point'
    polygon:
      type: array
      items:
        $ref: '#/components/schemas/line'
    ordinate:
      type: number
      example: -89.5327
  
```

### Describe the bug you're encountering

A GeoJSON object is a variable container which may provide a point, a line, a polygon or a bounding box.  A point is a (short) array of numbers.  A line is an array of points and a polygon is an array of lines (rings).  Each item builds upon the others.  The above definition has worked for years (even though the rendered example is not that compelling) but something recently has broken the system (perhaps related to 7040?) 
### To reproduce...

Copy and paste the schema above into https://editor.swagger.io/.  

Console log shows 

swagger-editor-bundle.js:sourcemap:2 TypeError: Cannot read property 'anyOf' of undefined
    at e (swagger-editor-bundle.js:sourcemap:2)
    at G (swagger-editor-bundle.js:sourcemap:2)
    at e (swagger-editor-bundle.js:sourcemap:2)
    at G (swagger-editor-bundle.js:sourcemap:2)
    at e (swagger-editor-bundle.js:sourcemap:2)
    at S (swagger-editor-bundle.js:sourcemap:2)
    at swagger-editor-bundle.js:sourcemap:2
    at ze (swagger-editor-bundle.js:sourcemap:2)
    at Ne (swagger-editor-bundle.js:sourcemap:2)
    at r.value (swagger-editor-bundle.js:sourcemap:2)

### Expected behavior
I expect swagger editor not to fail with a console error for valid openapi data.

### Screenshots
![capture](https://user-images.githubusercontent.com/4131067/110207561-48479180-7e52-11eb-8c15-d7222f6f20c6.png)

### Additional context or thoughts

Thanks for the help.

**Comment by mathis-m** [comment]
> The above definition has worked for years (even though the rendered example is not that compelling) but something recently has broken the system (perhaps related to 7040?)

@pauldzy do you now when it was introduced I stepped back to v3.39.0 but it does not work there either.

**Comment by pauldzy** [comment]
Hi Mathis,
US EPA has swagger-ui 3.38 running here which renders the schemas fine using the same geojson definition
https://watersgeo.epa.gov/openapi/waters/#/Navigation/x494

Mayhap it is 3.39 where it broke.

Paul

**Comment by mathis-m** [comment]
@pauldzy thanks for your quick response. I was able to solve the issue.

This was introduced with [v3.39.0](https://github.com/swagger-api/swagger-ui/releases/tag/v3.39.0) by my commit 0f541a1a.
There anyOf and oneOf was first introduced to the sample generator, unfortunately I did not handled the case were properties or items are defined via anyOf or oneOf.

This will be fixed with #7041. 

FYI: you were right about the correlation to #7040.

Best regards
Mathis
